doc/contribute/doc-guidelines.rst

```diff
@@ -0,0 +1,354 @@
+content::
+
+   The Makefile has targets that include:
+
+   html
+      Build the HTML output for the project
+
+   clean
+      Remove all generated output, restoring the folders to a
+      clean state.
+
+Would be rendered as:
+
+   The Makefile has targets that include:
+
+   html
+      Build the HTML output for the project
+
+   clean
+      Remove all generated output, restoring the folders to a
+      clean state.
+
+Multi-column lists
+******************
+
+If you have a long bullet list of items, where each item is short,
+you can indicate the list items should be rendered in multiple columns
+with a special ``hlist`` directive::
+
+   .. hlist::
+      :columns: 3
+
+      * A list of
+      * short items
+      * that should be
+      * displayed
+      * horizontally
+      * so it doesn't
+      * use up so much
+      * space on
+      * the page
+
+This would be rendered as:
+
+.. hlist::
+   :columns: 3
+
+   * A list of
+   * short items
+   * that should be
+   * displayed
+   * horizontally
+   * so it doesn't
+   * use up so much
+   * space on
+   * the page
+
+Note the optional ``:columns:`` parameter (default is two columns), and
+all the list items are indented by three spaces.
+
+File names and Commands
+***********************
+
+Sphinx extends reST by supporting additional inline markup elements (called
+"roles") used to tag text with special
+meanings and allow style output formatting. (You can refer to the `Sphinx Inline Markup`_
+documentation for the full list).
+
+For example, there are roles for marking :file:`filenames`
+(``:file:`name```) and command names such as :command:`make`
+(``:command:`make```).  You can also use the \`\`inline code\`\`
+markup (double backticks) to indicate a ``filename``.
+
+.. _internal-linking:
+
+Internal Cross-Reference Linking
+********************************
+
+ReST links are only supported within the current file using the
+notation::
+
+   refer to the `internal-linking`_ page
+
+which renders as,
+
+   refer to the `internal-linking`_ page
+
+Note the use of a trailing
+underscore to indicate an outbound link. In this example, the label was
+added immediately before a heading, so the text that's displayed is the
+heading text itself.
+
+With Sphinx however, we can create
+link-references to any tagged text within the Zephyr Project documentation.
+
+Target locations within documents are defined with a label directive:
+
+   .. code-block:: rst
+
+      .. _my label name:
+
+Note the leading underscore indicating an inbound link.
+The content immediately following
+this label is the target for a ``:ref:`my label name```
+reference from anywhere within the Zephyr documentation.
+The label should be added immediataly before a heading so there's a
+natural phrase to show when referencing this label (e.g., the heading
+text).
+
+This is the same directive used to
+define a label that's a reference to an URL::
```

Comment:
I think "URL" is usually pronounced "you are el", so "a URL" rather than "an URL", but if project convention is to pronounce it like "earl", never mind.